### PR TITLE
optimize markAllAsRead when no expiration timer & fix read receipts

### PR DESCRIPTION
**Pull request by Bilb**
we basically do a single sql call to mark everything as read for that
conversation, force unreadCount to 0 and mention state to false, and
trigger read syncs if needed.

the optimization cannot work for conversation with expiration timer for
now

### CODE REVIEW

File: ts/models/conversation.ts [hunk 1]
```diff
@@ -1195,15 +1196,50 @@ export class ConversationModel extends Backbone.Model<ConversationAttributes> {
     }
   }
 
+  /**
+   * Mark everything as read efficiently if possible.
+   *
+   * For convos with a expiration timer enable, start the timer as of no.
```
Comment (yougotwill) [inline ts/models/conversation.ts:1202]: I believe this should be "as of now."

File: ts/models/conversation.ts [hunk 2]
```diff
@@ -1195,15 +1196,50 @@ export class ConversationModel extends Backbone.Model<ConversationAttributes> {
...
+    if (!this.get('expireTimer')) {
+      const allReadMessages = await Data.markAllAsReadByConversationNoExpiration(this.id);
+      this.set({ mentionedUs: false, unreadCount: 0 });
+      await this.commit();
+      if (allReadMessages.length) {
+        await this.sendReadReceiptsIfNeeded(uniq(allReadMessages));
+      }
+      return;
+    }
+
```
Comment (yougotwill) [inline ts/models/conversation.ts:1227]: Don't need this new line

File: ts/models/conversation.ts [hunk 3]
```diff
@@ -1195,15 +1196,50 @@ export class ConversationModel extends Backbone.Model<ConversationAttributes> {
...
+    providedOptions: { sendReadReceipts?: boolean; readAt?: number } = {}
+  ) {
     const lastReadTimestamp = this.lastReadTimestamp;
     if (newestUnreadDate < lastReadTimestamp) {
       return;
     }
 
-    const options = providedOptions || {};
-    defaults(options, { sendReadReceipts: true });
+    const defaultedReadAt = providedOptions?.readAt || Date.now();
```
Comment (yougotwill) [inline ts/models/conversation.ts:1241]: defaulted normally means failed. Do you mean default?

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/models/conversation.ts (hunk 2 above)
Comment (Bilb) [reply ts/models/conversation.ts:1227]: fixed

File: ts/models/conversation.ts (hunk 1 above)
Comment (Bilb) [reply ts/models/conversation.ts:1202]: fxied

File: ts/models/conversation.ts (hunk 3 above)
Comment (Bilb) [reply ts/models/conversation.ts:1241]: fixed

**Review by yougotwill — APPROVED** [review]
LGTM 🚀

**Review by yougotwill — APPROVED** [review]
LGTM 🚀

**Pull request merged**